libraries/Microsoft.Bot.Builder.Expressions.Parser/ExpressionEngine.cs

```diff
@@ -143,13 +160,40 @@ public override Expression VisitIdAtom([NotNull] ExpressionParser.IdAtomContext
 
             public override Expression VisitIndexAccessExp([NotNull] ExpressionParser.IndexAccessExpContext context)
             {
-                var instance = Visit(context.primaryExpression());
-                var index = Visit(context.expression());
-                return MakeExpression(ExpressionType.Element, instance, index);
+                Expression instance;
+                var property = Visit(context.expression());
+
+                if (context.primaryExpression() is ExpressionParser.ShorthandAtomContext shorthandAtom)
+                {
+                    var shorthandMark = shorthandAtom.GetText();
+
+                    if (!ShorthandPrefixMap.ContainsKey(shorthandMark))
+                    {
+                        throw new Exception($"{shorthandMark} is not a shorthand");
+                    }
+
+                    if (shorthandMark == "^")
+                    {
+                        return MakeExpression(ExpressionType.Callstack, property);
+                    }
+
+                    instance = this.Transform(AntlrParse(ShorthandPrefixMap[shorthandMark]));
+                    var expression = MakeExpression(ExpressionType.Element, instance, property);
+
+                    return shorthandMark == "@" ? MakeExpression(ExpressionType.SimpleEntity, expression) : expression;
+                }
+
+                instance = Visit(context.primaryExpression());
+                return MakeExpression(ExpressionType.Element, instance, property);
             }
 
             public override Expression VisitMemberAccessExp([NotNull] ExpressionParser.MemberAccessExpContext context)
             {
+                if (context.primaryExpression() is ExpressionParser.ShorthandAtomContext shorthandAtom)
+                {
+                    throw new Exception($"Shorthand is not support with such format: {context.GetText()}");
```

Comment:
It would be good to suggest the right format, rather than just say this is not working